app/models/manageiq/providers/embedded_ansible/automation_manager/machine_credential.rb

```diff
@@ -1,55 +1,89 @@
+      :name       => 'userid',
+      :id         => 'userid',
     },
-    :password => {
-      :type      => :password,
-      :label     => N_('Password'),
-      :help_text => N_('Password for this credential')
-    }
-  }.freeze
+    {
+      :component  => 'password-field',
+      :label      => N_('Password'),
+      :helperText => N_('Password for this credential'),
+      :name       => 'password',
+      :id         => 'password',
+      :type       => 'password',
+    },
+  ].freeze
 
   # rubocop:disable Layout/AlignHash
   #
   # looks better to align the nested keys to the same distance, instead of
   # scope just for the hash in question (which is what rubocop does.
-  EXTRA_ATTRIBUTES = {
-    :ssh_key_data => {
-      :type       => :password,
-      :multiline  => true,
-      :label      => N_('Private key'),
-      :help_text  => N_('RSA or DSA private key to be used instead of password')
+  EXTRA_ATTRIBUTES = [
+    {
+      :component      => 'password-field',
+      :label          => N_('Private key'),
+      :helperText     => N_('RSA or DSA private key to be used instead of password'),
+      :componentClass => 'textarea',
+      :name           => 'ssh_key_data',
+      :id             => 'ssh_key_data',
+      :type           => 'password',
     },
-    :ssh_key_unlock => {
-      :type       => :password,
+    {
+      :component  => 'password-field',
       :label      => N_('Private key passphrase'),
-      :help_text  => N_('Passphrase to unlock SSH private key if encrypted'),
-      :max_length => 1024
+      :helperText => N_('Passphrase to unlock SSH private key if encrypted'),
+      :name       => 'ssh_key_unlock',
+      :id         => 'ssh_key_unlock',
+      :maxLength  => 1024,
+      :type       => 'password',
     },
-    :become_method => {
-      :type       => :choice,
-      :label      => N_('Privilege Escalation'),
-      :help_text  => N_('Privilege escalation method'),
-      :choices    => ['', 'sudo', 'su', 'pbrun', 'pfexec', 'doas', 'dzdo', 'pmrun', 'runas', 'enable', 'ksu', 'sesu', 'machinectl']
+    {
+      :component        => 'select',
+      :label            => N_('Privilege Escalation'),
+      :helperText       => N_('Privilege escalation method'),
+      :name             => 'become_method',
+      :id               => 'become_method',
+      :type             => 'choice',
+      :isClearable      => true,
+      :options          => [
+        {:label => 'sudo', :value => '1'},
+        {:label => 'su', :value => '2'},
+        {:label => 'pbrum', :value => '3'},
+        {:label => 'pfexec', :value => '4'},
+        {:label => 'doas', :value => '5'},
+        {:label => 'dzdo', :value => '6'},
+        {:label => 'pmrun', :value => '7'},
+        {:label => 'runas', :value => '8'},
+        {:label => 'enable', :value => '9'},
+        {:label => 'ksu', :value => '10'},
+        {:label => 'sesu', :value => '11'},
+        {:label => 'machinectl', :value => '12'},
```

Comment:
Why the numeric values, is it coming from the original angular implementation somehow?

Reply:
If I didn't add the values the label would come up but I wouldn't be able to select it

Reply:
That's fine, but are the values coming from the angular implementation or are they just your idea? If it's the latter, I think you can just have the same thing as label/value and write it [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself):
```ruby
%w[
  sudo
  su
  pbrum
  pfexec
  doas
  dzdo
  pmrun
  runas
  enable
  ksu
  sesu
  machinectl
].map { |item| { :label => item, :value => item } }
```

Reply:
yeah I should be able to re-write it like that